### PR TITLE
Extend committee `create-hot-key-authorization-certificate` to support scripts

### LINKED ISSUE #577: [FR] - Extend committee create-hot-key-authorization-certificate cmd to support scripts
Currently we have support **only** for key-based CC members:

```
cardano-cli conway governance committee create-hot-key-authorization-certificate
                                                                                          ( --cold-verification-key STRING
                                                                                          | --cold-verification-key-file FILE
                                                                                          | --cold-verification-key-hash STRING
                                                                                          )
                                                                                          ( --hot-key STRING
                                                                                          | --hot-key-file FILE
                                                                                          | --hot-key-hash STRING
                                                                                          )
                                                                                          --out-file FILE
```

We need to extend` committee create-hot-key-authorization-certificate ` to support script based CC members. 

conway cddl:
`auth_committee_hot_cert = (14, committee_cold_credential, committee_hot_credential)

### PR DESCRIPTION
**Pull request by carbolymer**
# Changelog

```yaml
- description: |
    Extend committee `create-hot-key-authorization-certificate` to support scripts
# uncomment types applicable to the change:
  type:
   - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
   - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
   - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Requires:
* https://github.com/IntersectMBO/cardano-api/pull/476

# How to trust this PR

Trust no one.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

**Comment by carbolymer** [comment]
PR isn't building, because it requires changes in API from: https://github.com/IntersectMBO/cardano-api/pull/476

File: cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_committee_create-hot-key-authorization-certificate.cli [hunk 1]
```diff
@@ -18,8 +20,11 @@ Available options:
                            Filepath of the Consitutional Committee cold key.
   --cold-verification-key-hash STRING
                            Constitutional Committee key hash (hex-encoded).
+  --cold-verification-key-script FILE
```
Comment (CarlosLopezDeLara) [inline cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_committee_create-hot-key-authorization-certificate.cli]: --cold-script-file        Cold Native or Plutus script file

File: cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_committee_create-hot-key-authorization-certificate.cli [hunk 2]
```diff
@@ -18,8 +20,11 @@ Available options:
...
+                           Cold key script
   --hot-key STRING         Constitutional Committee hot key (hex-encoded).
   --hot-key-file FILE      Filepath of the Consitutional Committee hot key.
   --hot-key-hash STRING    Constitutional Committee key hash (hex-encoded).
+  --hot-script FILE        Hot key script
```
Comment (CarlosLopezDeLara) [inline cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_committee_create-hot-key-authorization-certificate.cli]: --hot-script-file          Hot Native or Plutus script file

File: cardano-cli/test/cardano-cli-golden/files/golden/help.cli [hunk 3]
```diff
@@ -6346,10 +6346,12 @@ Usage: cardano-cli conway governance committee create-hot-key-authorization-cert
                                                                                           ( --cold-verification-key STRING
                                                                                           | --cold-verification-key-file FILE
                                                                                           | --cold-verification-key-hash STRING
+                                                                                          | --cold-verification-key-script FILE
```
Comment (CarlosLopezDeLara) [inline cardano-cli/test/cardano-cli-golden/files/golden/help.cli]: --cold-script-file

File: cardano-cli/test/cardano-cli-golden/files/golden/help.cli [hunk 4]
```diff
@@ -6346,10 +6346,12 @@ Usage: cardano-cli conway governance committee create-hot-key-authorization-cert
...
                                                                                           )
                                                                                           ( --hot-key STRING
                                                                                           | --hot-key-file FILE
                                                                                           | --hot-key-hash STRING
+                                                                                          | --hot-script FILE
```
Comment (CarlosLopezDeLara) [inline cardano-cli/test/cardano-cli-golden/files/golden/help.cli]: --hot-script-file

File: cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_committee_create-hot-key-authorization-certificate.cli [hunk 5]
```diff
@@ -2,10 +2,12 @@ Usage: cardano-cli conway governance committee create-hot-key-authorization-cert
                                                                                           ( --cold-verification-key STRING
                                                                                           | --cold-verification-key-file FILE
                                                                                           | --cold-verification-key-hash STRING
+                                                                                          | --cold-verification-key-script FILE
```
Comment (CarlosLopezDeLara) [inline cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_committee_create-hot-key-authorization-certificate.cli]: --cold-script-file

File: cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_committee_create-hot-key-authorization-certificate.cli [hunk 6]
```diff
@@ -2,10 +2,12 @@ Usage: cardano-cli conway governance committee create-hot-key-authorization-cert
...
                                                                                           )
                                                                                           ( --hot-key STRING
                                                                                           | --hot-key-file FILE
                                                                                           | --hot-key-hash STRING
+                                                                                          | --hot-script FILE
```
Comment (CarlosLopezDeLara) [inline cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_committee_create-hot-key-authorization-certificate.cli]: --hot-script-file

File: cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance/Committee.hs [hunk 7]
```diff
@@ -66,14 +66,14 @@ data GovernanceCommitteeCreateColdKeyResignationCertificateCmdArgs era =
     } deriving Show
 
 renderGovernanceCommitteeCmds :: GovernanceCommitteeCmds era -> Text
-renderGovernanceCommitteeCmds = \case
+renderGovernanceCommitteeCmds = ("governance committee" <>) . \case
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance/Committee.hs]:
```suggestion
renderGovernanceCommitteeCmds = ("governance committee " <>) . \case
```

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs [hunk 8]
```diff
@@ -250,7 +250,7 @@ pScriptFor name Nothing help' =
 
 pScriptFor name (Just deprecated) help' =
       pScriptFor name Nothing help'
-  <|> ScriptFile <$> Opt.strOption
+  <|> File <$> Opt.strOption
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs:253]: Why this change?

Comment (carbolymer) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs:253]: Because `ScriptFile` is redundant

File: cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance/Committee.hs (hunk 7 above)
Comment (carbolymer) [reply cardano-cli/src/Cardano/CLI/EraBased/Commands/Governance/Committee.hs]: good catch!

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs (hunk 8 above)
Comment (smelc) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs:253]: But it was meant to disambiguate a file containing anything from a file containing a script right? And this distinction still makes sense I think.

Anyway it's a small thing, feel free to ignore and resolve.

**Review by smelc — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs (hunk 8 above)
Comment (carbolymer) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs:253]: That's the purpose of a type parameter of `File`. `type ScriptFile = File ScriptInAnyLang In`.

File: cardano-cli/src/Cardano/CLI/Read.hs [hunk 9]
```diff
@@ -420,17 +422,35 @@ readScriptDataOrFile (ScriptDataCborFile fp) = do
...
 -- | Read a script file. The file can either be in the text envelope format
 -- wrapping the binary representation of any of the supported script languages,
 -- or alternatively it can be a JSON format file for one of the simple script
 -- language versions.
 --
-readFileScriptInAnyLang :: FilePath
-                        -> ExceptT (FileError ScriptDecodeError) IO
-                                   ScriptInAnyLang
+readFileScriptInAnyLang
+  :: MonadIOTransError (FileError ScriptDecodeError) t m
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Read.hs:448]: This constraint is overkill. We only call this in the context of `ExceptT`.

**Review by Jimbo4350 — COMMENTED** [review]
LGTM however I don't buy into `MonadIOTransError` at the moment.
